### PR TITLE
Allows middle-click and ctrl+click outbound clicks to properly open as user intended

### LINKED ISSUE #451: Outbound link tracking breaks shortcuts to Open in New Tab
## Prerequisites
- [x] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
**Describe the bug**
When I use the outbound link tracking version of the Plausible script, outbound links no longer open in a new tab when I Ctrl-Click or Cmd-Click them.

Instead, the current tab navigates to the destination after the event fires, the same as if I had clicked the link without any modifier keys.

**Expected behavior**
Normally, when I Ctrl-Click or Cmd-Click a link in my browser, it opens in a new tab. I expect outbound links on my site to behave this way too, even when Plausible is tracking the clicks.

**Environment (If applicable):**
 - OS: Windows and macOS
 - Browser: Firefox
 - Version: 83.0b9

### PR DESCRIPTION
**Pull request by Vigasaurus**
### Changes

Allows middle clicks or ctrl+click, cmd+click, shift+click to track to plausible, and open in a new tab if requested, instead of always redirecting the current tab. 

Fixes #451 

This will almost certainly cause conflicts with #489 in the compiled script, so I didn't push them in this PR at all. Once both are merged we can re-compile the merged version. 

### Tests
- [X] This PR does not require tests

### Changelog
- [x] Entry has been added to changelog

### Documentation
- [X] This change does not need a documentation update

### CODE REVIEW

File: tracker/src/plausible.js [hunk 1]
```diff
@@ -56,23 +56,34 @@
   }
 
   {{#if outboundLinks}}
-  function registerOutboundLinkEvents() {
-    document.addEventListener('click', function (event) {
-      var link = event.target;
+  function handleOutbound(event) {
+    var link = event.target;
+    var middle = event.type == "auxclick" && event.which == 2;
+    var click = event.type == "click";
```
Comment (ukutaht) [inline tracker/src/plausible.js:62]: What's going on with the spacing here? Seems like this line is 1 space to the left of the next line which cannot be correct.

File: tracker/src/plausible.js [hunk 2]
```diff
@@ -56,23 +56,34 @@
...
 
       if (link && link.href && link.host && link.host !== location.host) {
+        if (middle || click)
         plausible('Outbound Link: Click', {props: {url: link.href}})
 
         // Delay navigation so that Plausible is notified of the click
         if(!link.target || link.target.match(/^_(self|parent|top)$/i)) {
-          setTimeout(function() { location.href = link.href; }, 150);
-          event.preventDefault();
+          if (middle || event.ctrlKey || event.metaKey || event.shiftKey) window.open(link.href)
```
Comment (ukutaht) [inline tracker/src/plausible.js:73]: What does this line do? When someone uses middle click on `cmd+click`, I would expect we do nothing here and allow default browser behaviour.

Comment (ukutaht) [reply tracker/src/plausible.js:73]: Oh, the if-else statement is formatted inconsistently

**Review by ukutaht — CHANGES_REQUESTED** [review]
Nice way to account for middle clicks as well. Looks like formatting is a bit off and I'm curious what the purpose of `window.open` is when middle click or modifier is used.

**Comment by Vigasaurus** [comment]
So the reason for using window.open is that browsers interpret middle clicks as "Open in New Tab", the same as ctrl+click or cmd+click - and in the old way we'd always redirect the current tab to the new link. This way ensures that if a user does any of the actions that would normally redirect their click into a new tab (middle click, ctrl+click, cmd+click, or shift+click) - we open a new tab for the link as expected, as opposed to redirecting the current tab.

**Review by ukutaht — APPROVED** [review]

**Pull request merged**